integration/kubernetes/helm-chart/alluxio/values.yaml

```diff
@@ -310,3 +310,47 @@ fuse:
 #     alluxio-hdfs-config: hdfsConfig
 #   worker: # Shared by worker and jobWorker containers
 #     alluxio-hdfs-config: hdfsConfig
+
+# Remote logging server
+logserver:
+  enabled: true
+  replicas: 1
+  env:
+  # Extra environment variables for the logserver pod
+  # Example:
+  # JAVA_HOME: /opt/java
+  args: # Arguments to Docker entrypoint
+    - logserver
+  # Properties for the logserver component
+  properties:
+  resources:
+    # The default xmx is 8G
+    limits:
+      cpu: "4"
+      memory: "8G"
+    requests:
+      cpu: "1"
+      memory: "1G"
+  ports:
+    logging: 45600
+  hostPID: false
+  hostNetwork: false
+  # dnsPolicy will be ClusterFirstWithHostNet if hostNetwork: true
+  # and ClusterFirst if hostNetwork: false
+  # You can specify dnsPolicy here to override this inference
+  # dnsPolicy: ClusterFirst
+  # JVM options specific to the logserver container
+  jvmOptions:
+  nodeSelector: {}
+  # volumeType controls the type of log volume.
+  # It can be "persistentVolumeClaim" or "hostPath"
+  volumeType: persistentVolumeClaim
+  size: 4Gi
+  # Attributes to use if the log volume is PVC
+  pvcName: alluxio-logserver-logs
+  accessModes:
+    - ReadWriteOnce
+  storageClass: standard
+  # Attributes to use if the log volume is hostPath
+  hostPath: "/tmp/alluxio-logs" # The hostPath directory to use
```

Comment:
What if I do not want to use any additional volume for logging as my ephermal storage is enough for my requirement, then I do not see any option here. I just want to run a remote logging server without any additional volume.

Reply:
Could you elaborate a bit how you intend to provision your ephemeral storage for the logserver? In other words, i'm curious how much disk is given when you spawn a pod for the log server. 

If you are thinking about test usages where provisioning a PV would be overhead or a user, does supporting `emptyDir` make sense here?

Reply:
@jiacheliu3 I was talking about not adding any volume at all. I just want to run logserver based on given memory (or may be spilling to disk) but not want to attach any volume for storing logs.

```
spec.containers[].resources.limits.ephemeral-storage
spec.containers[].resources.requests.ephemeral-storage
```
Ref: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#configurations-for-local-ephemeral-storage

Reply:
@nirav-chotai Sure you do have control to the ephemeral storage, but I wouldn't recommend it because I feel it contradicts (at least partly) with why we want a logserver. To me the benefits of a log server are:
1. We have a centralized location for logs
2. We hold the logs when pods die

If we have the logs on the logserver but never persist that, we can still lose the logs. The logserver may be evicted for whatever reason, and you lose the logs. In the helm chart, the recommended (default) config will force the users to create and use PVs in multiple places. That was intended because we want the users to think about persistence for things like journal and worker's tiered storage. The same idea would apply to logs.

From that perspective, I guess my bottom line would be: it might be okay to allow using no volume at all, but that should be explicitly turned off. In other words, using a volume should be the default to be fool-proof.

Maybe you have your considerations or your design on how to use the logserver in your architecture? What are the benefits for you to only use ephemeral storage for the log server?

Reply:
Yes, you can enforce the default to have volume but there should be a way to use logserver without volume. 

My use case, in dev like environments I would like to run separate logserver so that whenever I rollout my changes (frequent for dev), it will persist good amount of logs in the logserver (based on the capacity of the logserver pod) while recreating only master and worker pods. Production like environments I am pretty sure anyone would go for PVC only. 

IMO, anyone would try and test in dev first and if things work, then they configure PVC (not necessarily available in all environments) to meet production requirements.

Reply:
And yes, your both points are valid even if I deploy logserver without volume

We have a centralized location for logs (yes)
We hold the logs when pods die (yes, logs will be there though master and workers pod die and re-created).

Reply:
Sorry for the late reply. I see your point and that use case is legit. I will add the option to not use a volume for the log server.

Reply:
@nirav-chotai I can do as my last [commit ](https://github.com/Alluxio/alluxio/pull/12662/commits/6213d9dc0e26f38bf04879d524faf833aa026cd3) to add a `none` option that creates no volume. But if `emptyDir` can just serve, I lean towards not adding another switch.

Reply:
I think I forgot to add `emptyDir` to the available options. Initially there was only `pvc` and `hostPath`. If there were only `pvc` and `hostPath`, I think it makes much sense to ask for sth that does need a volume like `none`. But if `emptyDir` is available, I think setting this option to `emptyDir` will serve your need.

Reply:
@jiacheliu3 Sorry, I was not available in the last few days. Yes, `emptyDir` will serve the need.

Reply:
Cool, will remove option `none`